operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/reconciler/dependent/managed/ManagedDependentResourceContext.java

```diff
@@ -1,97 +1,20 @@
 package io.javaoperatorsdk.operator.api.reconciler.dependent.managed;
 
-import java.util.Map;
 import java.util.Optional;
-import java.util.concurrent.ConcurrentHashMap;
 
-import io.javaoperatorsdk.operator.api.reconciler.dependent.DependentResource;
-import io.javaoperatorsdk.operator.api.reconciler.dependent.ReconcileResult;
+import io.javaoperatorsdk.operator.processing.dependent.workflow.WorkflowCleanupResult;
+import io.javaoperatorsdk.operator.processing.dependent.workflow.WorkflowReconcileResult;
 
-/**
- * Contextual information related to {@link DependentResource} either to retrieve the actual
- * implementations to interact with them or to pass information between them and/or the reconciler
- */
-@SuppressWarnings("rawtypes")
-public class ManagedDependentResourceContext {
+public interface ManagedDependentResourceContext {
```

Comment:
I extracted it for the same reason, since this is a user facing interface, so the setters should not be accessible. Thus is denotes the intention as much as possible.